src/library/scala/collection/IndexedSeq.scala

```diff
@@ -91,11 +91,25 @@ trait IndexedSeqOps[+A, +CC[_], +C] extends Any with SeqOps[A, CC, C] { self =>
 
   override def slice(from: Int, until: Int): C = fromSpecific(new IndexedSeqView.Slice(this, from, until))
 
-  override def head: A = apply(0)
+  override def head: A =
```

Comment:
wouldn't this now lead to repeated checks for `isEmpty` in `{head,last}Option`? should there be an `@inline def headImpl: A = apply(0)` and used in both `head` and `headOption`?

Reply:
My comment on the first commit is that head is fast. `headOption` is allocating.